x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java

```diff
@@ -185,7 +185,8 @@ static ClusterState moveClusterStateToStep(String indexName, ClusterState curren
 
         Step nextStep = stepRegistry.getStep(indexPolicySetting, nextStepKey);
         if (nextStep == null) {
-            throw new IllegalArgumentException("step [" + nextStepKey + "] with policy [" + indexPolicySetting + "] does not exist");
+            // stepRegistry may not be up-to-date with latest policies/steps in cluster-state
+            return currentState;
```

Comment:
Sorry, I had not looked closely enough at what this method is used for. It seems that its only used for the "move to step" and retry APIs so actually its probably ok for this method to throw an exception directly. I had thought before that we use this method for moving between steps normally.

I think we should throw directly here as the alternative of returning the same cluster state and then using that as an indication that the step is not recognised is a bit trappy. To ensure this method is kept only for API calls and not for normal execution maybe we can add a JavaDoc comment?

Reply:
OK. I will revert the latest commit and add comment